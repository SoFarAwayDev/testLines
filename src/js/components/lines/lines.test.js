import 'jsdom-global/register';
import React from 'react';

import Immutable from "immutable";
import { Provider } from "react-redux";
import configureStore from 'redux-mock-store';

import Lines from './lines.component';
import OneLine from './oneLine/oneLine.component';

const store = configureStore();

const mockStore = store({
	lines: Immutable.fromJS([
        {
            id: 0, name: 'redux line'
        },
        {
            id: 1, name: 'react line'  
        },
        {
            id: 2, name: 'immutable line'
        }
    ]),
});



describe('<Lines />', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = mount(
            <Provider store={mockStore}>
                <Lines />
            </Provider>);
    });

    it('lines match the snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('Should be 3 lines by default', () => {
        expect(wrapper.find(OneLine)).toHaveLength(3);
    });
});