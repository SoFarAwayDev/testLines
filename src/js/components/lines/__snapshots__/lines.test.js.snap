// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Lines /> lines match the snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Lines)>
    <Lines
      addLineAction={[Function]}
      deleteLineAction={[Function]}
      lines={
        Array [
          Object {
            "id": 0,
            "name": "redux line",
          },
          Object {
            "id": 1,
            "name": "react line",
          },
          Object {
            "id": 2,
            "name": "immutable line",
          },
        ]
      }
    >
      <div>
        <div>
          <div>
            <span>
              v 
              TEST
            </span>
          </div>
          <h2>
             Lines application (Massive Attack inspired)
          </h2>
          <div>
            <input
              onChange={[Function]}
              value=""
            />
            <button
              className="btn third"
              onClick={[Function]}
            >
              Add Line
            </button>
          </div>
          <div>
            <OneLine
              key="0"
              line={
                Object {
                  "id": 0,
                  "name": "redux line",
                }
              }
              onDeleteLine={[Function]}
            >
              <div>
                <span>
                  redux line
                </span>
                <i
                  className="material-icons md-24"
                  onClick={[Function]}
                >
                  clear
                </i>
              </div>
            </OneLine>
            <OneLine
              key="1"
              line={
                Object {
                  "id": 1,
                  "name": "react line",
                }
              }
              onDeleteLine={[Function]}
            >
              <div>
                <span>
                  react line
                </span>
                <i
                  className="material-icons md-24"
                  onClick={[Function]}
                >
                  clear
                </i>
              </div>
            </OneLine>
            <OneLine
              key="2"
              line={
                Object {
                  "id": 2,
                  "name": "immutable line",
                }
              }
              onDeleteLine={[Function]}
            >
              <div>
                <span>
                  immutable line
                </span>
                <i
                  className="material-icons md-24"
                  onClick={[Function]}
                >
                  clear
                </i>
              </div>
            </OneLine>
          </div>
        </div>
      </div>
    </Lines>
  </Connect(Lines)>
</Provider>
`;
